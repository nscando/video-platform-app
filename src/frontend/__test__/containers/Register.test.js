import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Register from '../../containers/Register';

describe('<Register/>', () => {

  test('Register form', () => {
    const preventDefault = jest.fn();
    const register = mount(
      <ProviderMock>
        <Register />
      </ProviderMock>,
    );

    register.find('form').simulate('submit', { preventDefault });
    expect(preventDefault).toHaveBeenCalledTimes(1);
    register.unmount();
  });

  test('Register Snapshot', () => {
    const register = create(
      <ProviderMock>
        <Register />
      </ProviderMock>,
    );
    expect(register.toJSON()).toMatchSnapshot();
  });

});
