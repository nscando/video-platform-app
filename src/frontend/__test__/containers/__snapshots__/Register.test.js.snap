// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register/> Register Snapshot 1`] = `
Array [
  <header
    className="header isRegister"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Logo"
        className="header__img"
        src="test-file-stub"
      />
    </a>
    <div
      className="header__menu"
    >
      <div
        className="header__menu--profile"
      >
        <img
          alt="userIcon"
          src="test-file-stub"
        />
        <p>
          Perfil
        </p>
      </div>
      <ul>
        <li>
          <a
            href="/login"
            onClick={[Function]}
          >
            Iniciar Sesion
          </a>
        </li>
      </ul>
    </div>
  </header>,
  <section
    className="register"
  >
    <section
      className="register__container"
    >
      <h2>
        Regístrate
      </h2>
      <form
        className="register__container--form"
        onSubmit={[Function]}
      >
        <input
          className="input"
          name="name"
          onChange={[Function]}
          placeholder="Nombre"
          type="text"
        />
        <input
          className="input"
          name="email"
          onChange={[Function]}
          placeholder="Correo"
          type="text"
        />
        <input
          className="input"
          name="password"
          onChange={[Function]}
          placeholder="Contraseña"
          type="password"
        />
        <button
          className="button"
          type="submit"
        >
          Registrarme
        </button>
      </form>
      <a
        href="/login"
        onClick={[Function]}
      >
        Iniciar sesión
      </a>
    </section>
  </section>,
]
`;
